### PR TITLE
Implement kick function

### PR DESCRIPTION
**Pull request by OTCompa**
Follow up of PR #1, implements the basic functionality for the `/kick` function.
Original PR: https://github.com/OTCompa/moddingway/pull/1

### CODE REVIEW

File: internal/discord/command-handlers.go [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+// CheckUserInGuild checks if the user is in the specified server.
+func (d *Discord) CheckUserInGuild(gid string, user string) error {
+	_, err := d.Session.GuildMember(gid, user)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *Discord) CommandLogger(i *discordgo.Interaction) error {
```
Comment (brtran4) [inline internal/discord/command-handlers.go:63]: same thing here, let's add a doc string. let's also change this func name to `LogCommand`

File: internal/discord/commands.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+		var guildname string
+		guild, err := d.Session.Guild(i.GuildID)
+		if err != nil {
+			fmt.Printf("Unable to find guild name: %v", err)
+			guildname = i.GuildID
+		} else {
+			guildname = guild.Name
+		}
+
+		tempstr := fmt.Sprintf("You are being kicked from %v for the reason:\n%v",
```
Comment (brtran4) [inline internal/discord/commands.go:64]: can you add screenshots of the kick message here to see what it looks like?

File: internal/discord/commands.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not kick user <@%v>", userToKick)
+		fmt.Printf("%v: %v\n", tempstr, err)
+
+		err = StartInteraction(s, i.Interaction, tempstr)
+		if err != nil {
+			fmt.Printf("Unable to send ephemeral message: %v\n", err)
+		}
+	} else {
```
Comment (brtran4) [inline internal/discord/commands.go:101]: i think it makes more sense to move this up in the func. how i picture it is:
1. user uses the command and fills out the fields
2. we dm the user
3. we kick them from the server

**Review by brtran4 — CHANGES_REQUESTED** [review]
some commits from your first PR are present in this PR. is it possible to take them out of this PR?

File: internal/discord/commands.go (hunk 2 above)
Comment (OTCompa) [reply internal/discord/commands.go:68]: image

**Comment by OTCompa** [comment]
The above commits edits the resulting log message and also sends a followup message whether the user was able to be DMed or not

File: internal/discord/commands.go (hunk 3 above)
Comment (OTCompa) [reply internal/discord/commands.go]: That's the same order unless I'm mistaken.
With the changes below, it should be something along these lines:
1. User uses command
2. Bot logs usage of command 
3. Bot attempts to DM user
4. Bot kicks user regardless 
5. If kicked user was not able to be DMed, inform invoker and edit log message to reflect it

File: internal/discord/commands.go (hunk 2 above)
Comment (brtran4) [reply internal/discord/commands.go:68]: ok cool this looks good

File: internal/discord/commands.go [hunk 4]
```diff
@@ -1,16 +1,126 @@
...
+
+	// DM the user regarding the kick
+	channel, err := s.UserChannelCreate(userToKick)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not create a DM with user %v", userToKick)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		dmFailed = true
+	} else {
+		// Get guild name
+		// We could cut down on an API call by hardcoding this
```
Comment (brtran4) [inline internal/discord/commands.go:58]: let's make a constant file and define the the hardcoded `GUILDNAME` there, then import into this file and other files that we may need in the future

File: internal/discord/commands.go (hunk 3 above)
Comment (brtran4) [reply internal/discord/commands.go]: ah i got error output confused with actual input text. yeah this looks good to me

**Review by brtran4 — CHANGES_REQUESTED** [review]

File: internal/discord/command-handlers.go [hunk 5]
```diff
@@ -32,6 +36,7 @@ func (d *Discord) AddCommands(s *discordgo.Session, event *discordgo.Ready) {
 			PurgeCommand,
 			ExileCommand,
 			UnexileCommand,
+			SetModLogCommand,
```
Comment (brtran4) [inline internal/discord/command-handlers.go:39]: arguably we should probably move this to its own PR and merge this in first

File: internal/discord/command-handlers.go [hunk 6]
```diff
@@ -46,6 +51,71 @@ func (d *Discord) AddCommands(s *discordgo.Session, event *discordgo.Ready) {
 	}
 }
 
+// CheckUserInGuild checks if the user is in the specified server.
+func (d *Discord) CheckUserInGuild(gid string, user string) error {
```
Comment (brtran4) [inline internal/discord/command-handlers.go:55]: let's type the full name and do `guild_id`

File: internal/discord/command-handlers.go [hunk 7]
```diff
@@ -46,6 +51,71 @@ func (d *Discord) AddCommands(s *discordgo.Session, event *discordgo.Ready) {
...
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// LogCommand logs the moderation command in the channel specified by LogChannelID
+// It sends an embed with all command arguments as separate fields
+// It additionally returns the sent message in case any edits need to be made
+func (d *Discord) LogCommand(i *discordgo.Interaction) (*discordgo.Message, error) {
```
Comment (brtran4) [inline internal/discord/command-handlers.go:66]: let's double check that it does not send ephemeral messages

**Review by brtran4 — CHANGES_REQUESTED** [review]
